concrete/src/File/FileListTransformer.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Concrete\Core\File;
+
+use League\Fractal\TransformerAbstract;
+
+class FileListTransformer extends TransformerAbstract
+{
+    public function transform(FileList $fileList)
+    {
+        $results = [];
+
+        foreach ($fileList->getPagination()->getCurrentPageResults() as $file) {
```

Comment:
$list->getPagination() is deprecated, add the `\Concrete\Core\Search\Pagination\PaginationFactory` as a dependency and use it to get the pagination instance instead

Reply:
Pulling this file out; this is from the derek pull request

Reply:
I'm not a fan of the FileListTransformer, if i was doing it now id probably re-write it, it could be a collection of Files with the Pagination Adapter added to the resource.